### PR TITLE
[4.2] (re) Introduce a body end renderer

### LINKED ISSUE #37297: [4.x] Modal dialogs don't appear if there are two editors on a page in different tabs when using Editor - None/Editor - CodeMirror
### Steps to reproduce the issue
1. Set default editor to Editor - None or Editor - CodeMirror
2. Create a custom module
3. In the .xml file, create two separate `<fieldset>` with a `<field type="editor" />` each
4. Open the module in the backend to edit, swap to the second tab and try to open any editor extensions (ie. Media)

### Expected result
The modal dialog appears

### Actual result
The modal backdrop appears, but the modal dialog does not.

### Additional comments
When inspecting the DOM elements, the issue appears to be that the modal dialog elements are placed in the first tab, and since tabs have a `display:none` when inactive, the dialog itself is also hidden.

I tested with TinyMCE, something about how that one is setup/instantiated has it either fix itself or just handle the modal dialogs differently.

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request is the base for many Issues associated with modals. Fixes #37297

### Summary of Changes
- Introduces another renderer
- Adds setter/getter for the enqueued html fragments

#### Backend Toolbars

Try to edit an article and check the source of the document (scroll to the end) [preferably with debug disabled]
Apply the PR
Check again the source, the last entries before < /body > should be some modals
Check that the toolbar buttons versions, preview and a11y checks are all functionals

#### Backend Multiple editors in different tabs #37297

Ensure the global editor is set to 'None'
Edit `administrator/components/com_content/forms/article.xml` and add 

```xml
<field name="xoxo" type="editor" editor="none" buttons="true" />
```

After line 264. Try to check if both editors have functional buttons

Apply the PR and repeat the test

Repeat the test for code mirror

### Actual result BEFORE applying this Pull Request

Chunks of HTML can only be echoed sequentially, eg the toolbar modals are echoed immediately after the button 

Screenshot 2022-03-20 at 15 13 24

### Expected result AFTER applying this Pull Request

Chunks of HTML can be enqueued to be echoed at the end of the document

Screenshot 2022-03-20 at 15 12 41

### Documentation Changes Required

@laoneo this is needed if you want to see any improvement on the modals

@roland-d in order to solve https://github.com/joomla/joomla-cms/issues/37297 and a few more modal related issues this (or something similar) is required

### CODE REVIEW

File: libraries/src/Document/Document.php [hunk 1]
```diff
@@ -262,6 +262,14 @@ class Document
 	 */
 	protected $webAssetManager = null;
 
+		/**
```
Comment (brianteeman) [inline libraries/src/Document/Document.php]:
```suggestion
	/**
```

**Comment by dgrammatiko** [comment]
There's nothing in their docs: https://www.php.net/manual/en/function.serialize.php
But probably I could do this in another way

**Comment by alikon** [comment]
FYI https://php.watch/versions/8.1/serializable-deprecated

**Comment by dgrammatiko** [comment]
@brianteeman @alikon that article is talking about implementing an own serialised/unserializer and that for 8.1 it needs both the old and the new functions. But I'm not using `implements` here. So is this an issue here?

File: layouts/joomla/editors/buttons.php [hunk 2]
```diff
@@ -16,7 +16,7 @@
 ?>
 <div class="editor-xtd-buttons" role="toolbar" aria-label="<?php echo Text::_('JTOOLBAR'); ?>">
 	<?php if ($buttons) : ?>
-		<?php foreach ($buttons as $button) : ?>
+		<?php foreach (array_map('unserialize', array_unique(array_map('serialize', $buttons))) as $button) : ?>
```
Comment (laoneo) [inline layouts/joomla/editors/buttons.php]: Is this really needed like that?

Comment (dgrammatiko) [reply layouts/joomla/editors/buttons.php]: The problem is that button get added as the event fires multiple times and the results are accumulating. Probably this is the wrong solution, the right thing is to fix the particular event to return **only** the buttons for the requested call or don't use the event system at all for the buttons...

Comment (laoneo) [reply layouts/joomla/editors/buttons.php]: ok

File: libraries/src/Document/Renderer/Html/BodyendRenderer.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	 *
+	 * @param   string  $name     Not used.
+	 * @param   array   $params   Associative array of values
+	 * @param   string  $content  Not used.
+	 *
+	 * @return  string  The output of the script
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function render($name, $params = array(), $content = null)
```
Comment (Quy) [inline libraries/src/Document/Renderer/Html/BodyendRenderer.php]:
```suggestion
	public function render($name, $params = [], $content = null)
```

File: plugins/editors/none/none.php [hunk 4]
```diff
@@ -91,7 +91,7 @@ public function _displayButtons($name, $buttons, $asset, $author)
 			$buttonsEvent = new Event(
 				'getButtons',
 				[
-					'editor'  => $name,
+					'editor'    => $name,
```
Comment (Quy) [inline plugins/editors/none/none.php]: This is no longer aligned.

Comment (dgrammatiko) [reply plugins/editors/none/none.php]: I know but it's irrelevant to this PR

**Comment by dgrammatiko** [comment]
@SharkyKZ why the thumbs down?

**Comment by laoneo** [comment]
Why did you close?

**Pull request closed without merging**